oak-store-document/src/test/java/org/apache/jackrabbit/oak/plugins/document/MongoDocumentStoreTest.java

```diff
@@ -241,6 +244,19 @@ public void queryWithLimit() throws Exception {
         store.dispose();
     }
 
+    @Test(expected = DocumentStoreException.class)
+    public void queryWithNodeNameTooLong() {
+        DocumentStore docStore = openDocumentStore();
+        DocumentNodeStore store = builderProvider.newBuilder().setDocumentStore(docStore).setAsyncDelay(0).getNodeStore();
+        Revision rev = Revision.newRevision(0);
+        List<UpdateOp> inserts = Lists.newArrayList();
+        for (int i = 0; i < 2; i++) {
+            DocumentNodeState n = new DocumentNodeState(store, Path.fromString(DocumentMK.LONG_PATH + i),
+                    new RevisionVector(rev));
+            inserts.add(n.asOperation(rev));
+        }
```

Comment:
Same thing as with the other test : it expects to fail but doesn't assert that explicitly (just implicitly that the test itself should throw an exception). Might be matter of taste, but this would be an alternative:
```suggestion
        DocumentNodeState n = new DocumentNodeState(store, Path.fromString(DocumentMK.LONG_PATH + i),
                new RevisionVector(rev));
        inserts.add(n.asOperation(rev));
        fail("should not reach this point");
```

Reply:
(actually my example doesn't compile I just noticed - the `i` has to be defined)

Reply:
(but outside of the existance of a loop that `i` maybe doesn't make much sense neither so could also be removed i guess)